lib/beaker/hypervisor/aws_sdk.rb

```diff
@@ -687,7 +687,7 @@ def key_name_prefix
     # @return [String] safe key name for current host
     # @api private
     def key_name
-      "#{key_name_prefix}-#{@options[:timestamp].strftime("%F_%H_%M_%S")}"
+      "#{key_name_prefix}-#{@options[:aws_keyname_modifier]}-#{@options[:timestamp].strftime("%F_%H_%M_%S")}"
```

Comment:
Why not just add a random string instead? That way we (and anyone else using aws, who knows) don't have to change the job configuration. Since this bug is a regression from original behaviour, its a less user-impacting way of solving it.